plutus-scb/src/Plutus/SCB/Events/Contract.hs

```diff
@@ -1,92 +1,216 @@
   deriving  (Eq, Show, Generic)
   deriving anyclass (FromJSON, ToJSON)
 
-type ContractRequestEvent = RequestEvent ContractRequest
+instance Pretty ContractRequest where
+    pretty = \case
+        AwaitSlotRequest s -> "AwaitSlot:" <+> pretty s
+        AwaitTxConfirmedRequest t -> "AwaitTxConfirmed:" <+> pretty t
+        UserEndpointRequest t -> "UserEndpoint:" <+> pretty t
+        OwnPubkeyRequest r -> "OwnPubKey:" <+> pretty r
+        UtxoAtRequest a -> "UtxoAt:" <+> pretty a
+        NextTxAtRequest a -> "NextTxAt:" <+> pretty a
+        WriteTxRequest u -> "WriteTx:" <+> pretty u
 
 data ContractResponse =
   AwaitSlotResponse Slot
-  | AwaitTxConfirmedResponse TxId
-  | UserEndpointResponse Value
+  | AwaitTxConfirmedResponse TxConfirmed
+  | UserEndpointResponse Text (EndpointValue Value)
   | OwnPubkeyResponse PubKey
   | UtxoAtResponse UtxoAtAddress
-  | NextTxAtResponse Tx
+  | NextTxAtResponse (Address, Tx)
   | WriteTxResponse W.WriteTxResponse
   deriving  (Eq, Show, Generic)
   deriving anyclass (FromJSON, ToJSON)
 
-type ContractResponseEvent = ResponseEvent ContractResponse
+instance Pretty ContractResponse where
+    pretty = \case
+        AwaitSlotResponse s        -> "AwaitSlot:" <+> pretty s
+        UserEndpointResponse n r     -> "UserEndpoint:" <+> pretty n <+> pretty r
+        OwnPubkeyResponse pk       -> "OwnPubKey:" <+> pretty pk
+        UtxoAtResponse utxo        -> "UtxoAt:" <+> pretty utxo
+        NextTxAtResponse r         -> "NextTxAt:" <+> pretty r
+        WriteTxResponse w          -> "WriteTx:" <+> pretty w
+        AwaitTxConfirmedResponse w -> "AwaitTxConfirmed:" <+> pretty w
+
+data ContractInstanceState t =
+    ContractInstanceState
+        { csContract           :: ContractInstanceId
+        , csCurrentIteration   :: ContractIteration
+        , csCurrentState       :: PartiallyDecodedResponse
+        , csContractDefinition :: t
+        }
+    deriving (Show, Eq, Generic)
+    deriving anyclass (ToJSON, FromJSON)
+
+instance Pretty t => Pretty (ContractInstanceState t) where
+    pretty ContractInstanceState{csContract, csCurrentIteration, csCurrentState, csContractDefinition} =
+        hang 2 $ vsep
+            [ "Instance:" <+> pretty csContract
+            , "Iteration:" <+> pretty csCurrentIteration
+            , "State:" <+> pretty csCurrentState
+            , "Contract definition:" <+> pretty csContractDefinition
+            ]
+
+data PartiallyDecodedResponse =
+    PartiallyDecodedResponse
+        { newState :: Value
+        , hooks    :: Value
+        }
+    deriving (Show, Eq, Generic)
+    deriving anyclass (ToJSON, FromJSON)
+
+instance Pretty PartiallyDecodedResponse where
+    pretty PartiallyDecodedResponse {newState, hooks} =
+        vsep
+            [ "State:"
+            , indent 2 $ pretty $ take 120 $ BS8.unpack $ JSON.encodePretty newState
+            , "Hooks:"
+            , indent 2 $ pretty $ take 120 $ BS8.unpack $ JSON.encodePretty hooks
```

Comment:
Two things here:

1) We've got `Plutus.SCB.Utils.abbreviate` which will do a prettier job than just `take 120`.
2) When I use `plutus-scb contracts status <uuid>` I usually want to see the whole thing.

Reply:
I'll abbreviate the `state` and print the full `hooks` value. The start of the state is enough to see if the contract is waiting for input and the rest of the state is pretty verbose and doesn't mean that much.